frontend/src/utils/handleCaptionsSignal/handleCaptionsSignal.ts

```diff
@@ -0,0 +1,63 @@
+import { Dispatch, SetStateAction } from 'react';
+import { SignalEvent } from '../../types/session';
+
+/**
+ * @typedef {object} CaptionsSignalDataType
+ * @property {string} action - The action to be performed on captions ('enable', 'disable').
+ * @property {string} captionsId - The ID of the captions to be enabled or disabled.
+ */
+export type CaptionsSignalDataType = {
+  action: 'enable' | 'disable';
+  captionsId: string;
+};
+
+/**
+ * @typedef {object} CaptionsHandleType
+ * @property {SignalEvent} event - The signal event containing the data.
+ * @property {Dispatch<SetStateAction<boolean>>} setIsSessionCaptioningEnabled - Function to set the captions enabled state.
+ */
+export type CaptionsHandleType = {
+  event: SignalEvent;
+  setIsSessionCaptioningEnabled: Dispatch<SetStateAction<boolean>>;
```

Comment:
Do we receive a signal when joining a session? This might be out-of-sync if we don't, otherwise.

Reply:
but I think for consistency sake, it'd be good to have the `isSessionCaptioningEnabled` to be the same across all clients. I'm going to make some changes to the `joinRoom` flow that will include captions. SIgnaling when joining a session seems to not play well in the case of joining the room